modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2251,6 +2251,23 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
         labels="drop",
     )
 
+    # __setitem__ methods
+    def setitem_bool(self, row_loc, col_loc, item):
+        def _set_item(df, row_loc):
+            df = df.copy()
+            df.loc[row_loc.squeeze(axis=1), col_loc] = item
+            return df
+
+        new_modin_frame = self._modin_frame.broadcast_apply_full_axis(
```

Comment:
do we need full-axis apply though? Can we just use simple `.broadcast_apply` here that would map to all of the partitions?

Reply:
regarding `.broadcast_apply`: we may also want to disable this co-partition step by some flag indicating that we're ensured that the arguments' partitioning is aligned:
https://github.com/modin-project/modin/blob/7c1cb3c96aa1a9123fa9edb544a1cdcbe34a41f2/modin/core/dataframe/pandas/dataframe/dataframe.py#L2465-L2472

Reply:
We do `broadcast_apply_full_axis` because we need to know all columns of the frame in order to correctly set an item.

Reply:
ah, because we can also filter along columns, got it,

Reply:
Columns may be in different column partitions.